common.sh

```diff
@@ -160,9 +175,247 @@ exit_fixed()
+                "passwd")
+                    if [ ! -f "$PREUPG_PASSWD" ]; then
+                        log_error "File $PREUPG_PASSWD does not exist. It is mandatory."
+                    else
+                        cat $PREUPG_PASSWD
+                    fi
+                    ;;
+                "group")
+                    if [ ! -f "$PREUPG_GROUP" ]; then
+                        log_error "File $PREUPG_GROUP does not exist. It is mandatory."
+                    else
+                        cat $PREUPG_GROUP
+                    fi
+                    ;;
+                *)
+                    log_error "Unknown $key for $command. Supported are 'passwd', 'group'."
+                    return 1
+                    ;;
+            esac
+            ;;
+        "command")
+            case $key in
+                "rpm_qa")
+                    _get_cached_command $PREUPG_RPM_QA
+                    ;;
+                "rpm_Va")
+                    _get_cached_command $PREUPG_ALL_CHANGED
+                    ;;
+                "rpm_etc_Va")
+                    _get_cached_command $PREUPG_CONFIGCHANGED
+                    ;;
+                "chconfig")
+                    _get_cached_command $PREUPG_CHKCONFIG
+                    ;;
+                *)
+                    log_error "Unknown $key for $command. Supported are 'rpm_qa', 'rpm_Va', 'rpm_etc_Va', 'chkconfig'".
+                    return 1
+                    ;;
+            esac
+            ;;
+        "filelist")
+            case $key in
+                "allmyfiles")
+                    _get_cached_command $PREUPG_ALLMYFILES
+                    ;;
+                "executable")
+                    _get_cached_command $PREUPG_EXECUTABLES
+                    ;;
+                "rpmtrackedfiles")
+                    _get_cached_command $PREUPG_RPMTRACKEDFILES
+                    ;;
+                "rpmrhsignedfiles")
+                    _get_cached_command $PREUPG_RPM_RHSIGNED
+                    ;;
+                *)
+                    log_error "Unknown $key for $command. Supported are 'executable', 'allmyfiles', 'rpmtrackedfiles', 'rpmrhsignedfiles'."
+                    return 1
+            esac
+            ;;
+        "info")
+            ;;
+        *)
+            log_error "Unknown command $command. Supported are 'file', 'command', filelist', 'info'."
+            return 1
+            ;;
+    esac
+
+}
+
+add_postupgrade()
+{
+    # Add postupgrade script to /root/preupgrade/postupgrade.d directory
+    # File as first parameter has to exists in modules directory
+    # Usage:    add_postupgrade file
+
+    script_name=$1
+    postupgrade_name="$PREUPG_POSTUPGRADE_DIR/`basename $script_name`"
+    [ ! -f "$script_name" ] && log_error "$script_name does not exist." && exit_error
+    [ -f "$postupgrade_name" ] && log_warning "$script_name already exists in $PREUPG_POSTUPGRADE_DIR"
+    cp -af $script_name $postupgrade_name
+}
```

Comment:
Sure, but it was another Pull Request. And this Pull Request takes a pretty much time.